src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h

```diff
@@ -30,37 +31,46 @@ struct grpc_s2a_credentials_options {
  public:
   ~grpc_s2a_credentials_options();
 
-  /** Setter methods. They do not take ownership of the arguments, nor do they
-   * remove a duplicate ciphersuite or target service account from the
-   * appropriate vector, if a duplicate exists. **/
+  /** Getters for member fields. **/
+  const std::string handshaker_service_url() const {
+    return handshaker_service_url_;
+  }
+  const std::vector<uint16_t>& supported_ciphersuites() const {
+    return supported_ciphersuites_;
+  }
+  const std::vector<std::string>& target_service_account_list() const {
+    return target_service_account_list_;
+  }
+
   /** The setter method for |handshaker_service_url_|. This does not take
    *  ownership of the argument. **/
-  void set_handshaker_service_url(const char* handshaker_service_url);
+  void set_handshaker_service_url(std::string handshaker_service_url);
   /** This methods add |ciphersuite| to the vector |supported_ciphersuites_|; it
    *  does not remove duplicates from the vector, if they exist. See
    *  src/core/tsi/s2a/s2a_constants.h for the ciphersuite constants. **/
   void add_supported_ciphersuite(uint16_t ciphersuite);
   /** This API should only be called at the client-side, and any target service
    *  accounts that are added on the server-side will be ignored. This method
-   * adds a target service account to the vector |target_service_account_list_|;
-   * it does not remove duplicates from the vector, nor does it take ownership
-   * of the argument. **/
-  void add_target_service_account(const char* target_service_account);
+   *  adds a target service account to the vector
+   *  |target_service_account_list_|; it does not remove duplicates from the
+   *  vector, nor does it take ownership of the argument. **/
+  void add_target_service_account(std::string target_service_account);
 
   /** Create a deep copy of this grpc_s2a_credentials_options instance. **/
   grpc_s2a_credentials_options* copy() const;
 
   /** This method returns true if the fields of this
    *  grpc_s2a_credentials_options instance match the arguments; otherwise, it
    *  returns false. It is used only for testing purposes. **/
-  bool check_fields(const char* handshaker_service_url,
-                    const std::vector<uint16_t>& supported_ciphersuites,
-                    const std::vector<char*>& target_service_account_list);
+  bool check_fields(
```

Comment:
Is this function used any place other than tests? Do we need to expose it  as a public method?

Reply:
It is for testing purposes only. I can call it `CheckFieldsForTesting` instead?